src/modules/hip/kernel/blend.cpp

```diff
@@ -1,67 +1,115 @@
 #include <hip/hip_runtime.h>
-#define saturate_8u(value) ( (value) > 255 ? 255 : ((value) < 0 ? 0 : (value) ))
 
-__device__ unsigned char blend_formula( unsigned char input_pixel1, unsigned char input_pixel2, float alpha){
+#if defined(STATIC)
+#include "rpp_hip_host_decls.hpp"
+#endif
+
+#define saturate_8u(value) ((value) > 255 ? 255 : ((value) < 0 ? 0 : (value)))
+
+__device__ unsigned char blend_formula(unsigned char input_pixel1, unsigned char input_pixel2, float alpha)
+{
     return saturate_8u(alpha * input_pixel1 + (1 - alpha) * input_pixel2);
 }
 
-extern "C" __global__ void blend(     unsigned char* input1,
-                         unsigned char* input2,
-                         unsigned char* output,
-                        const unsigned int height,
-                        const unsigned int width,
-                        const float alpha,
-                        const unsigned int channel
-)
+extern "C" __global__ void blend(unsigned char *input1,
+                                 unsigned char *input2,
+                                 unsigned char *output,
+                                 const unsigned int height,
+                                 const unsigned int width,
+                                 const float alpha,
+                                 const unsigned int channel)
 {
-    int id_x = hipBlockIdx_x *hipBlockDim_x + hipThreadIdx_x;
-    int id_y = hipBlockIdx_y *hipBlockDim_y + hipThreadIdx_y;
-    int id_z = hipBlockIdx_z *hipBlockDim_z + hipThreadIdx_z;
-    if (id_x >= width || id_y >= height || id_z >= channel) return;
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if (id_x >= width || id_y >= height || id_z >= channel)
+    {
+        return;
+    }
 
     int pixIdx = id_x + id_y * width + id_z * width * height;
 
     output[pixIdx] = ((1-alpha) * input1[pixIdx]) + (alpha * input2[pixIdx]);
```

Comment:
need to rewrite this taking advantage of float vector computation

Reply:
We'll do the kernel level optimizations subsequent PRs